src/jrd/replication/Publisher.cpp

```diff
@@ -226,74 +244,100 @@ namespace
 	}
 
 	class ReplicatedRecordImpl :
-		public AutoIface<IReplicatedRecordImpl<ReplicatedRecordImpl, CheckStatusWrapper> >
+		public AutoIface<IReplicatedRecordImpl<ReplicatedRecordImpl, CheckStatusWrapper> >,
+		public IReplicatedFieldImpl<ReplicatedRecordImpl, CheckStatusWrapper>
 	{
 	public:
-		ReplicatedRecordImpl(thread_db* tdbb, const Record* record)
+		ReplicatedRecordImpl(thread_db* tdbb, const jrd_rel* relation, const Record* record)
 			: //m_tdbb(tdbb),
-			  m_record(record)
+			  m_record(record),
+			  m_format(record->getFormat()),
+			  m_relation(relation)
 		{
 		}
 
 		~ReplicatedRecordImpl()
 		{
 		}
 
-		unsigned getRawLength()
+		unsigned getCount() override
 		{
-			return m_record->getLength();
+			return m_format->fmt_count;
 		}
```

Comment:
It would require code in getField() scanning whole field list from the beginning on each call. It shall turns O(N) into O(N^2) for record data handling. Unacceptable price for convenience, IMHO.

Reply:
I see no sense in returning something that cannot be used. As for performance, scanning could be done just once, in the constructor. A mapping table will be required, of course, but it could be made HalfStaticArray with large enough (e.g. 256) initial capacity and maybe even reused between multiple records via the Replicator object. Alternatively, scanning could be done per every field but starting with the prior m_fieldIndex (m_position should also be saved). The most common (sequential) field access pattern will perform with O(N).

Reply:
I still think that this is a needless complication. Check for getField() returning nullptr has to be in every plugin in any case so skipping of disappeared fields is easy there. At least in my plugin I would prefer to have a real field id (just in case if I decide to use it for fields mapping instead of name).

Reply:
Actually, now, when I said it, I realized that my existing code using field names will fail on renamed fields and that's a problem...

Reply:
OK, I can live with that. It would be up to plugin to handle missing fields.